src/sonic-eventd/src/eventd.cpp

```diff
@@ -547,8 +548,9 @@ capture_service::set_control(capture_control_t ctrl, event_serialized_lst_t *lst
         case INIT_CAPTURE:
             m_thr = thread(&capture_service::do_capture, this);
             for(int i=0; !m_cap_run && (i < CAPTURE_SERVICE_POLLING_RETRIES); ++i) {
-                /* Wait max a second for thread to init */
-                this_thread::sleep_for(chrono::milliseconds(CAPTURE_SERVICE_POLLING_DURATION));
+                /* Poll to see if thread has been init, if so exit early. Add delay on every attempt */
```

Comment:
>Poll [](http://example.com/codeflow?start=19&length=4)

It reduces the possibility of the issue happening, and it is not a completed fix. Then we need to handle the issue as a warning, not as an error.

Reply:
I added logic that eventd will not exit when capture service fails to initialize; warning logs are added, and main proxy/zmq service will continue to run.

Reply:
closed.